apps/accounts/views.py

```diff
@@ -38,3 +52,25 @@ class ProfileView(JWTProtectedView, generics.RetrieveAPIView):
 
     def retrieve(self, request, *args, **kw):
         return Response(self.get_serializer(request.user).data)
+
+
+class RegisterView(APIView):
+
+    @waffle_switch('fxa-auth')
+    def post(self, request):
+        if 'code' not in request.DATA:
+            return Response({'error': 'No code provided.'}, status=422)
+
+        try:
+            identity = verify.fxa_identify(request.DATA['code'],
+                                           config=settings.FXA_CONFIG)
+        except verify.IdentificationError:
+            return Response({'error': 'Profile not found.'}, status=401)
```

Comment:
The code above looks exactly like the one in `LoginView.post`, do you want to extract it in a helper like `validate_post` or something, which returns the identity?